kernels/keep_m_in_n.cu

```diff
@@ -0,0 +1,116 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2021 Mark Bauer
+ *
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ *
+ */
+
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+#include <cusp/keep_m_in_n.cuh>
+#include <complex>
+
+namespace cusp {
+
+template <typename T>
+__global__ void kernel_keep_m_in_n(const T *in, T *out, int m, int window, int N) {
```

Comment:
Because this kernel is not performing any arithmetic operations, it can be implemented without templating and just operate on items of a particular size - this would require additional work, so I'll still merge without it and it can be done later